packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -0,0 +1,490 @@
+									}
+									confirmAction={ () => {
+										onCommentResolve( thread.id );
+										setActionState( false );
+									} }
+									discardAction={ () =>
+										setActionState( false )
+									}
+								/>
+							) }
+						{ 'delete' === actionState?.action &&
+							thread.id === actionState?.id && (
+								<ConfirmNotice
+									confirmMessage={
+										// translators: message displayed when deleting a comment
+										__(
+											'Are you sure you want to delete this thread?'
+										)
+									}
+									confirmAction={ () => {
+										onCommentDelete( thread.id );
+										setActionState( false );
+									} }
+									discardAction={ () =>
+										setActionState( false )
+									}
+								/>
+							) }
+						{ 0 < thread?.reply?.length &&
+							thread.reply.map( ( reply ) => (
+								<VStack
+									key={ reply.id }
+									className="editor-collab-sidebar-panel__child-thread"
+									id={ reply.id }
+									spacing="2"
+								>
+									<CommentBoard
+										thread={ reply }
+										parentThread={ thread }
+									/>
+									{ 'delete' === actionState?.action &&
+										reply.id === actionState?.id && (
+											<ConfirmNotice
+												confirmMessage={
+													// translators: message displayed when deleting a comment
+													__(
+														'Are you sure you want to delete this thread?'
+													)
+												}
+												confirmAction={ () => {
+													onCommentDelete( reply.id );
+													setActionState( false );
+												} }
+												discardAction={ () =>
+													setActionState( false )
+												}
+											/>
+										) }
+								</VStack>
+							) ) }
+					</VStack>
+				) ) }
+		</>
+	);
+}
+
+/**
+ * EditComment component.
+ *
+ * @param {Object}   props          - The component props.
+ * @param {Object}   props.thread   - The comment thread object.
+ * @param {Function} props.onUpdate - The function to call when updating the comment.
+ * @param {Function} props.onCancel - The function to call when canceling the comment update.
+ * @return {JSX.Element} The EditComment component.
+ */
+function EditComment( { thread, onUpdate, onCancel } ) {
+	const [ inputComment, setInputComment ] = useState(
+		removep( thread.content.rendered )
+	);
+
+	return (
+		<>
+			<TextareaControl
+				__nextHasNoMarginBottom
+				className="editor-collab-sidebar-panel__comment-field-textarea"
+				value={ inputComment ?? '' }
+				onChange={ setInputComment }
+			/>
+			<VStack alignment="left" spacing="3" justify="flex-start">
+				<HStack alignment="left" spacing="3" justify="flex-start">
+					<Button
+						__next40pxDefaultSize
+						accessibleWhenDisabled
+						variant="primary"
+						onClick={ () => onUpdate( inputComment ) }
+						size="compact"
+						disabled={
+							0 === sanitizeCommentString( inputComment ).length
+						}
+					>
+						{ _x( 'Update', 'Update comment' ) }
```

Comment:
Using `verb` or `noun` usually helps more when specifying context.

```suggestion
						{ _x( 'Update', 'verb' ) }
```